applets/wncklet/window-list.c

```diff
@@ -319,7 +320,23 @@ static void applet_size_allocate(GtkWidget *widget, GtkAllocation *allocation, T
 
 	g_assert(len % 2 == 0);
 
-	mate_panel_applet_set_size_hints(MATE_PANEL_APPLET(tasklist->applet), size_hints, len, 0);
+	/* When loading the WnckTasklist initially, it reports size hints as though there were no
+	 * elements in the Tasklist. We keep a flag to skip size hinting until WnckTasklist has
+	 * something to show. */
```

Comment:
How this is a problem? Are not size hints updated when there are new elements? I forgot how this works...

Reply:
The size hints are updated. WnckTasklist works just fine, but I think there's a race condition in how we set those hints initially that is only apparent with GDK_SCALE=2. Doing this here defers that initial size hint until the applet is fully loaded.

Reply:
You think or you have found something that proves that? Why would this work differently with GDK_SCALE=2?

What size hints you get with scale == 1? What numbers you get with scale == 2?

Reply:
I have not yet found proof of this. WnckTasklist works fine regardless of GDK_SCALE - it loads size hints as `{0, 0}` initially, then updates to `{0, 402}` (in my system with my screen resolution, etc). That's all good. The problem seems to be that the applet is built out-of-process, and in the time between when the first `{0, 0}` hint and the second `{0, 402}` hint are sent, something happens that gets the applet locked into a non-scaled state. This doesn't happen with in-process applet, nor with GDK_SCALE=1. So the issue is clearly somewhere else, but this code here sidetracks it until we can fix whatever causes this in libmate-panel-applet.

Maybe I'll add a `/* HACK: ... */` message to make this clear.

Reply:
I've updated this code to only run for out-of-process wncklet, and made the comment a bit more explicit.